src/deploymentTool.ts

```diff
@@ -0,0 +1,19 @@
+export const BASE = "cli-firebase";
+
+export function value() {
+  if (!process.env.FIREBASE_DEPLOY_AGENT) {
+    return BASE;
+  }
+
+  return [BASE, process.env.FIREBASE_DEPLOY_AGENT].join("--");
+}
+
+export function labels() {
+  return {
+    "deployment-tool": value(),
+  };
+}
+
+export function check(labels: { [key: string]: any }) {
+  return labels && labels["deployment-tool"] && labels["deployment-tool"].indexOf(BASE) === 0;
```

Comment:
nit: This might be slightly more readable as `labels["deployment-tool"].startsWith(BASE)` over `labels["deployment-tool"].indexOf(BASE) === 0`

Reply:
Could also use optional chaining:

```suggestion
  return labels?.["deployment-tool"]?.startsWith(BASE);
```